SIGs/GRC/publications/Confidential Payload Upgrade Governance.md

```diff
@@ -0,0 +1,46 @@
+
+# Problem
+
+Ensuring that the overall system – starting with Verifier and ending with Relying Parties – continues to operate without incurring undue downtime during the upgrade/rollback period requires careful orchestration.
+
+# Forces
+
+Critical distributed applications often have high availability requirements. In addition, Control Specifications or Control Objectives may dictate that for certain classes of problems (e.g., high-impact vulnerabilities), all systems must be updated within a given time interval, sometimes an aggressive one.
+
+# Solution
+
+The terms MUST/SHOULD/MAY etc. below are used in accordance with **\[2\]**. Every SHOULD recommendation is explained separately in the “SHOULD vs. MUST Clarifications” section towards the end of this document.
+
+Integrate the upgrade with the software deployment stack (e.g., the CI/CD pipeline).  
+Once a new Confidential Payload[^1] version becomes available, the Verifier policies[^2] SHOULD **\[a\]** allow both old and new versions of the Payload to be treated as valid. If the rollout of the new Payload version is successful, the old Payload version is then removed from the Verifier policies. Conversely, if the rollout experiences difficulties, a rollback is initiated and the new Payload version is removed from the Verifier policies leaving only the old Payload version as valid. This is illustrated in the diagram below.
+
+![Payload Update Solution Diagram](./images/payload_update_sol.png)
+
+# Governance Expectations Summary
+
+The numbers in the left column below refer to **\[3\]**. Rows listed as N/A indicate that corresponding expectations are listed under different Patterns documents.
+
+| \# | Description |
+| :---- | :---- |
+| 1-6, 8-15 | N/A |
+| 7 | Update Verifier policies with new expected Reference Values before deployment begins. Roll back this update if the Payload deployment fails for any reason. Remove Verifier policies related to older versions(s) of the Payload measurements after the deployment of updated Payload is successful. |
+
+# “SHOULD” vs. “MUST” Clarifications
+
+1. Failure to handle intermediate states where only a subset of Payload instances are upgraded may result in downtime or intermittent failures related to the upgrade – where only upgraded Payloads can successfully attest and existing Payloads that may need to reattest wouldn’t be able to.
```

Comment:
Simplify wording / address run-on

```suggestion
1. Failure to handle intermediate states where only a subset of Payload instances are upgraded could result in downtime or intermittent failures related to the upgrade, i.e., where only upgraded Payloads can successfully attest and existing Payloads will fail verification against the upgraded policies.
```

Reply:
"would fail" instead of "will fail"? Not sure what's the correct grammar here.

Reply:
I guess "would" is probably better - agrees in mood with could